octodns/provider/ns1.py

```diff
@@ -933,12 +933,19 @@ def _monitor_gen(self, record, value):
             request = r'GET {path} HTTP/1.0\r\nHost: {host}\r\n' \
                 r'User-agent: NS1\r\n\r\n'.format(path=path, host=host)
             ret['config']['send'] = request
-            # We'll also expect a HTTP response
-            ret['rules'] = [{
-                'comparison': 'contains',
-                'key': 'output',
-                'value': '200 OK',
-            }]
+            rules = ret.setdefault('rules', [])
+            if record.healthcheck_statuscode is not None:
```

Comment:
This is now allowing the config to contain an arbitrary status code, and if anything is defined it's adding `200 OK` to the list of requirements. If someone were to set `201` this wouldn't work as expected.

Relatedly since this capability is being added I took a look at how it would be implemented with Route53 and this bit is interesting:

> HTTP : Route 53 tries to establish a TCP connection. If successful, Route 53 submits an HTTP request and waits for an HTTP status code of 200 or greater and less than 400.

So it doesn't allow things to specify a status code as-is, but there is a way to include a single contains string via a `SearchString` parameter to the healthcheck config.

Since Route53 doesn't support specifying a status code and this implementation and this implementation doesn't match the schema you've defined I'd recommend either:

1. Dropping the ability to configure the status code and just leaving NS1 as-is with a `200 OK` always there and the response contain if that's configured. 
2. Implementing a lookup table in NS1 to map the status code to the expected string, e.g. `200: '200 OK'`,  `201: '201 Created'`, ...

I'd leant towards 1 based on Route53's lack of configurability and the fact that I'm not aware of anyone requesting/needing the ability to change the response status code.

Reply:
> 1. Dropping the ability to configure the status code and just leaving NS1 as-is with a `200 OK` always there and the response contain if that's configured.

This is doable except it doesn't solve for our use case where certain legacy endpoints monitored are network/firewall devices which do not really return a "200 OK" in the response. Instead it just returns a login form containing string such as `Welcome to Cisco` or `please enter administrator username` etc. In such cases multiple `UP` conditions having to match `200 OK` as well as `Welcome to` will not pass. 

I am going to look at those legacy devices to see if anything else can be monitored on them. But as it stands, do you think there is any other way to achieve this?

Reply:
OK. In that case there would probably need to be a translation between status code and the expected text, though it doesn't really line up with what Route53 does/supports. 

We could maybe include the `200 OK` if no `healthcheck_response_contains` is defined and if it is only do the contains, skipping the 200. That wouldn't line up exactly with Route53, which would still require `200<=status<400`, but that's a static requirement for it and it would at least would be consistent with what's there currently.

Reply:
@ross  I was able to make things work with existing implementation of checking for "200 OK". We can close this PR as "not needed" at this time.